app/services/variant_units/option_value_namer.rb

```diff
@@ -58,7 +58,7 @@ def option_value_value_unit
     def option_value_value_unit_scaled
       unit_scale, unit_name = scale_for_unit_value
 
-      value = (@nameable.unit_value / unit_scale).to_d.truncate(2)
+      value = (@nameable.unit_value / unit_scale).to_d.round(2)
```

Comment:
I'm not sure I fully understand what can happen here. Is it possible to add a spec ([eg here](https://github.com/openfoodfoundation/openfoodnetwork/blob/master/spec/services/variant_units/option_value_namer_spec.rb#L158)) to show why we need to? 

Looking at the history, it looks like both `to_d` (BigDecimal) and `truncate` were added to help deal with unexpected numbers from Float manipulation. 

I'm not sure, but maybe we should fix the root problem by casting the source values to BigDecimal first.
But we still need to round it though, so I agree to use `round`. If `round` is taking care of it all maybe BigDecimal isn't needed at all here?

Reply:
I agree. I guess that the correct solution would be:
```suggestion
      value = (@nameable.unit_value.to_d / unit_scale).round(2)
```